### PR TITLE
[server] remove user relation from gitpod_token

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

The relation would pull non-fully initialized user objects into the system. We should rather not use relations but keys and explicitly fetch the user when needed.

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - se-remove-relation</li>
	<li><b>🔗 URL</b> - <a href="https://se-remove-relation.preview.gitpod-dev.com/workspaces" target="_blank">se-remove-relation.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/gitpod-db/src/typeorm/entity/db-gitpod-token.ts [hunk 1]
```diff
@@ -24,9 +23,8 @@ export class DBGitpodToken implements GitpodToken {
     @Column({ type: "int" })
     type: GitpodTokenType;
 
-    @ManyToOne((type) => DBUser)
-    @JoinColumn()
```
[the remark this answers is not in the document]
Comment (geropl) [reply components/gitpod-db/src/typeorm/entity/db-gitpod-token.ts:28]: Do we have some basic tests that read/write tokens? Would be good to verify in code that this "migration" works smoothly.

File: components/gitpod-db/src/typeorm/entity/db-gitpod-token.ts [hunk 2]
```diff
@@ -24,9 +24,8 @@ export class DBGitpodToken implements GitpodToken {
...
-    user: DBUser;
+    @Column(TypeORM.UUID_COLUMN_TYPE)
+    userId: string;
```
Comment (svenefftinge) [inline components/gitpod-db/src/typeorm/entity/db-gitpod-token.ts:28]: This is the main change, i.e. we remove the ORM relationship mapping and replace it with a simple foreign key exposure.

**Review by selfcontained — APPROVED** [review]
LGTM - tested logging in and a few things in the preview environment as well.

**Pull request merged**